lib/recursive-methods.rb

```diff
@@ -1,49 +1,137 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) with n being how high the value of the integer is
+# Space complexity: O(n), because of the recursive stack; size of stack depends on value of parameter
 def factorial(n)
-    raise NotImplementedError, "Method not implemented"
+  if n < 0 
+    raise ArgumentError.new('Provide a number greater than 0.')
+  end
+  return 1 if n == 0
+
+  return (n * factorial(n - 1))
 end
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), with n being the length of the string parameter
+# Space complexity: O(n^2), because of the recursive stack; size of the stack depends on the length of the string.
+# There is an additional array structure also created to aid in reversal of the string.
 def reverse(s)
```

Comment:
Since you're not creating a new array with **each** recursive call, this has space complexity O(n)